code/modules/mob/living/carbon/xenomorph/castes/puppet/puppet.dm

```diff
@@ -0,0 +1,53 @@
+// THIS SHOULD NEVER BE IN PLAYER CONTROL (like actually having a client)
+/mob/living/carbon/xenomorph/puppet
+	caste_base_type = /mob/living/carbon/xenomorph/puppet
+	name = "Puppet"
+	desc = "A reanimated body, crudely pieced together and held in place by an ominous energy tethered to some unknown force."
+	icon = 'icons/Xeno/1x1_Xenos.dmi'
+	icon_state = "Puppet Running"
+	health = 150
+	maxHealth = 150
+	plasma_stored = 0
+	pixel_x = 0
+	old_x = 0
+	tier = XENO_TIER_MINION
+	upgrade = XENO_UPGRADE_BASETYPE
+	pull_speed = -1
+	allow_pass_flags = PASS_XENO
```

Comment:
@bowlofcereal are puppets suppossed to go through each other/xenos

Reply:
> @bowlofcereal are puppets suppossed to go through each other/xenos

yeah its to avoid a clumps of puppets blocking xenos from being able to retreat and getting killed because of it